vllm/model_executor/models/h2ovl.py

```diff
@@ -0,0 +1,400 @@
+            ((i // (target_width // image_size)) + 1) * image_size,
+        )
+        # split the image
+        split_img = resized_img.crop(box)
+        processed_images.append(split_img)
+    assert len(processed_images) == blocks
+    if use_thumbnail and len(processed_images) != 1:
+        thumbnail_img = image.resize((image_size, image_size))
+        processed_images.append(thumbnail_img)
+    return processed_images, target_aspect_ratio
+
+
+# new dynamic_preprocess2 with prior_aspect_ratio
+def dynamic_preprocess2(
+    image: Image.Image,
+    min_num: int,
+    max_num: int,
+    image_size: int,
+    use_thumbnail: bool,
+    prior_aspect_ratio: Tuple[int, int],
+) -> List[Image.Image]:
+    orig_width, orig_height = image.size
+
+    # calculate the number of blocks based on prior aspect ratio
+    blocks, target_width, target_height, _ = calculate_num_blocks(
+        orig_width,
+        orig_height,
+        min_num,
+        max_num,
+        image_size,
+        use_thumbnail=False,
+        prior_aspect_ratio=prior_aspect_ratio,
+    )
+    # resize the image
+    resized_img = image.resize((target_width, target_height))
+    processed_images = []
+    for i in range(blocks):
+        box = (
+            (i % (target_width // image_size)) * image_size,
+            (i // (target_width // image_size)) * image_size,
+            ((i % (target_width // image_size)) + 1) * image_size,
+            ((i // (target_width // image_size)) + 1) * image_size,
+        )
+        # split the image
+        split_img = resized_img.crop(box)
+        processed_images.append(split_img)
+    assert len(processed_images) == blocks
+    if use_thumbnail and len(processed_images) != 1:
+        thumbnail_img = image.resize((image_size, image_size))
+        processed_images.append(thumbnail_img)
+    return processed_images
+
+
+def load_image1(image: Image.Image, input_size=448, min_num=1, max_num=6):
+    # image = Image.open(image_file).convert('RGB')
+    transform = build_transform(input_size=input_size)
+    images, target_aspect_ratio = dynamic_preprocess(
+        image,
+        image_size=input_size,
+        use_thumbnail=True,
+        min_num=min_num,
+        max_num=max_num,
+    )
+    pixel_values = [transform(image) for image in images]
+    pixel_values = torch.stack(pixel_values)
+    return pixel_values, target_aspect_ratio
+
+
+def load_image2(
+    image: Image.Image,
+    input_size=448,
+    min_num=1,
+    max_num=6,
+    target_aspect_ratio=None,
+):
+    # image = Image.open(image_file).convert('RGB')
+    transform = build_transform(input_size=input_size)
+    images = dynamic_preprocess2(
+        image,
+        image_size=input_size,
+        use_thumbnail=True,
+        min_num=min_num,
+        max_num=max_num,
+        prior_aspect_ratio=target_aspect_ratio,
+    )
+    pixel_values = [transform(image) for image in images]
+    pixel_values = torch.stack(pixel_values)
+    return pixel_values
```

Comment:
Seems that `load_image2` works just like `load_image1` if `target_aspect_ratio=None`, only differs from retrun values. Can we replace `load_image1` with `load_image2` with the modification of `dynamic_preprocess`?

Reply:
The duplicated implementation is fixed.